### PR TITLE
Backend securitygroups

### PR DESCRIPTION
**Pull request by Unicus-Tom**
I feel its a bit sloppy but unsure how to make it cleaner. Suggestions welcome in code review.

### CODE REVIEW

File: RestApi/Sepes.RestApi/Controllers/PodController.cs [hunk 1]
```diff
@@ -30,6 +31,37 @@ public async Task<Pod> createPod([FromBody] PodInput input)
...
+        public async Task removeNsg([FromBody] NsgModel data)
+        {
+
+            await _pod.removeNsg(data.resourceGroupName, data.subnetName, data.networkName);
+        }
+        [HttpDelete("nsg/purgeunused")]
+        public async Task<UInt16> deleteUnused()
+        {
+            return await _pod.deleteUnused();
+        }
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Controllers/PodController.cs]: Lets keep it out off the controller. If you want to test something you can create a DevController where you add temp and play-around APIes.

File: RestApi/Sepes.RestApi/Services/IAzureService.cs [hunk 2]
```diff
@@ -8,5 +8,12 @@ public interface IAzureService
         Task<string> CreateResourceGroup(string podName);
         Task TerminateResourceGroup(string resourceGroupName);
         Task<string> CreateNetwork(string networkName, string addressSpace);
+        Task CreateSecurityGroup(string securityGroupName, string resourceGroupName);
+        Task DeleteSecurityGroup(string securityGroupName, string resourceGroupName);
+        Task ApplySecurityGroup(string resourceGroupName, string securityGroupName, string subnetName, string networkName);
+        Task RemoveSecurityGroup(string resourceGroupName, string subnetName, string networkName);
+        Task NsgAllowInboundPort(string securityGroupName, string resourceGroupName, string ruleName, int priority, string[] externalAddresses, int externalPort);
+        Task NsgAllowOutboundPort(string securityGroupName, string resourceGroupName, string ruleName, int priority, string[] internalAddresses, int internalPort);
+        Task NsgAllowPort(string securityGroupName, string resourceGroupName, string ruleName, int priority, string[] internalAddresses, int internalPort, string[] externalAddresses, int externalPort);
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/IAzureService.cs]: Just a Note: Only put stuff in interface that will be used by the pod service.
This Interface looks a bit intimidating. Just a note don't need to fix now.

File: RestApi/Sepes.RestApi/Services/PodService.cs [hunk 3]
```diff
@@ -9,6 +10,11 @@ namespace Sepes.RestApi.Services
     public interface IPodService
     {
         Task<Pod> CreateNewPod(string name, int userID);
+        Task applyNsg(string resourceGroupName, string securityGroupName, string subnetName, string networkName);
+        Task removeNsg(string resourceGroupName, string subnetName, string networkName);
+        Task<UInt16> deleteUnused();
+        Task deleteNsg(string securityGroupName, string resourceGroupName);
+        Task createNsg(string securityGroupName, string resourceGroupName);
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/PodService.cs]: At least remove from interface for now.

**Review by gjersvik — CHANGES_REQUESTED** [review]
PodController and IPodService should be cleaned the rest are just hints and tips.

**Review by gjersvik — APPROVED** [review]

**Review by Kris2ffer — APPROVED** [review]

**Pull request merged**